service/controller/v2/resources/vaultcrt/current.go

```diff
@@ -49,6 +50,11 @@ func (r *Resource) GetCurrentState(ctx context.Context, obj interface{}) (interf
 	// draining was done and the pods got removed we get an empty list here after
 	// the delete event got replayed. Then we just remove the secrets as usual.
 	if key.IsDeleted(customObject) {
+		if !r.checkCertType(customObject) {
+			r.logger.LogCtx(ctx, "level", "debug", "message", "do not supporting this cert type any longer.")
+			return secret, nil
```

Comment:
Should the whole reconciliation or at least resource be cancelled?

Reply:
Since we need to keep going ahead with the deletion process, I don't think we could cancel any kind of things in here, neither reconciliation nor resource.